custom_components/unifiprotect/binary_sensor.py

```diff
@@ -57,67 +90,91 @@ async def async_setup_entry(
     if not protect_data.data:
         return
 
-    sensors = []
+    entities = []
     for device_id in protect_data.data:
         device_data = protect_data.data[device_id]
         if device_data["type"] == DEVICE_TYPE_DOORBELL:
-            sensors.append(
+
+            entities.append(
                 UnifiProtectBinarySensor(
                     upv_object,
                     protect_data,
                     server_info,
                     device_id,
-                    DEVICE_TYPE_DOORBELL,
-                    None,
+                    UnifiProtectBinaryEntityDescription(
+                        key=DEVICE_TYPE_DOORBELL,
+                        name=DEVICE_TYPE_DOORBELL,
+                        device_class=DEVICE_CLASS_OCCUPANCY,
+                        icon="mdi:doorbell-video",
+                        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+                        ufp_device_type=DEVICE_TYPE_DOORBELL,
+                    ),
                 )
             )
             _LOGGER.debug(
-                "UNIFIPROTECT DOORBELL SENSOR CREATED: %s",
+                "Adding Doorbel Binary Sensor entity for %s",
                 device_data["name"],
             )
 
         if device_data["type"] in DEVICES_WITH_CAMERA:
-            sensors.append(
+            entities.append(
                 UnifiProtectBinarySensor(
                     upv_object,
                     protect_data,
                     server_info,
                     device_id,
-                    DEVICE_TYPE_MOTION,
-                    None,
+                    UnifiProtectBinaryEntityDescription(
+                        key=DEVICE_TYPE_MOTION,
+                        name=DEVICE_TYPE_MOTION,
+                        device_class=DEVICE_CLASS_MOTION,
+                        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+                        ufp_device_type=DEVICE_TYPE_MOTION,
+                    ),
```

Comment:
Since these are all constants, the `UnifiProtectBinaryEntityDescription` could be a constant above.

Then you can build up a list of `UnifiProtectBinaryEntityDescription` and and create the entities in a loop since `UnifiProtectBinarySensor` will take the  same arguments except for the loop variable of entity description

something like (typing this on github so the indent will be wrong)

```
for entity_desc in entity_descs:
  entities.append(
                UnifiProtectBinarySensor(
                    upv_object,
                    protect_data,
                    server_info,
                    device_id,
                    entity_desc
                )
  )
```

... and if all the logic is simple and `entity_desc` is the only thing that changes in the loop, you could even use a list comprehension.

Reply:
I will give it a go. The only issue I see with this is with the DEVICE_TYPE_SENSOR as these are variable based on what is installed and activated. But with you suggestion I should at least be able to minimize the number of appends (If I can figure it out)

Reply:
I have been looking at the code now for 30 min. and I can't get my head around how to simplify this without getting to many binary sensors per physical device (Sensors that should not be attached to a Camera f.ex.) I understand what you want me to do, but how would I ensure that the DOORBELL sensor only gets added when the device is a doorbell?

Reply:
At the airport (again).  If wifi is working in flight I will see if I can work out some suggested code.

Reply:
@briisI pushed a change to implement what I described above.  I am still in flight so I'm not able to test it. 

I'm likely going to be exhausted when I get home so you will probably see it before I wake up and double check it.  I usually don't push code that probably doesn't work (don't have a proper dev setup in flight) but I didn't want you to be stuck, and I figure its close enough that it only needs few more tweaks to get it working correctly.

Reply:
Thanks for the suggestion @bdraco. I pulled this, and besides some minor things with a missing ( it looks good, but it does not work. I get the below error, which I cannot see why happens, as the key is present in all descriptions.

```log
2021-11-12 05:27:55 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up unifiprotect platform for binary_sensor
Traceback (most recent call last):
  File "/usr/local/python/lib/python3.9/site-packages/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/unifiprotect/binary_sensor.py", line 134, in async_setup_entry
    UnifiProtectBinarySensor(
  File "/config/custom_components/unifiprotect/binary_sensor.py", line 163, in __init__
    super().__init__(upv_object, protect_data, server_info, device_id, description)
  File "/config/custom_components/unifiprotect/entity.py", line 35, in __init__
    self._unique_id = f"{self._description.key}_{self._mac}"
AttributeError: 'tuple' object has no attribute 'key'
````

Reply:
Should be good now.  I had to fix a few things once I got home and got a live test environment 

I did some more cleanups to `number.py` [`605474c` (#332)](https://github.com/briis/unifiprotect/pull/332/commits/605474ca92ca58dedd957ab978a7a81ea86650ec) to use `.items()`

The other platforms could use the same but I'm too tired to even see the screen so I'm going to get some sleep

Reply:
I can confirm that it works now - Big thank you Nick. I will see what I can do with `.items()` on the other files. Sleep well.